lib/elixir/lib/enum.ex

```diff
@@ -1408,7 +1408,14 @@ defmodule Enum do
 
   """
   @spec sum(t) :: number
-  def sum(enumerable) do
+  def sum(first..last) when last > first do
+    ((last*(last+1))/2) - ((first*(first-1))/2)    
+  end
+  def sum(first..last) do # When first >= last
```

Comment:
I'd remove the `# When first >= last` comment as it's clear enough without it. If you want it there, I'd use the actual `when first >= last` guard (extremely small overhead anyways).